tap/tapify.py

```diff
@@ -1,106 +1,293 @@
 
-        # Get the default or required of the argument
         if param.name in func_kwargs:
-            tap_kwargs["default"] = func_kwargs[param.name]
+            is_required = False
+            default = func_kwargs[param.name]
             del func_kwargs[param.name]
-        elif param.default != Parameter.empty:
-            tap_kwargs["default"] = param.default
+        elif param.default != inspect.Parameter.empty:
+            is_required = False
+            default = param.default
         else:
-            tap_kwargs["required"] = True
+            is_required = True
+            default = inspect.Parameter.empty  # Can be set to anything. It'll be ignored
+
+        arg_data = _ArgData(
+            name=param_name,
+            annotation=annotation,
+            is_required=is_required,
+            default=default,
+            description=param_to_description.get(param.name),
+        )
+        args_data.append(arg_data)
+    return _TapData(args_data, has_kwargs, known_only)
+
+
+def _is_data_model(obj: Any) -> bool:
+    return dataclasses.is_dataclass(obj) or _is_pydantic_base_model(obj)
+
+
+def _docstring(class_or_function):
+    """Parse class or function docstring in one line"""
+    if inspect.isclass(class_or_function) and class_or_function.__init__.__doc__ is not None:
+        doc = class_or_function.__init__.__doc__
+    else:
+        doc = class_or_function.__doc__
+    return parse(doc)
+
+
+def _tap_data(class_or_function: _ClassOrFunction, docstring: Docstring, func_kwargs):
+    # Get the description of each argument in the class init or function
+    param_to_description = {param.arg_name: param.description for param in docstring.params}
+    if _is_data_model(class_or_function):
+        # TODO: allow passing func_kwargs to a Pydantic BaseModel
+        return _tap_data_from_data_model(class_or_function, func_kwargs, param_to_description)
+    else:
+        return _tap_data_from_class_or_function(class_or_function, func_kwargs, param_to_description)
+
+
+def _tap_class(args_data: Sequence[_ArgData]) -> Type[Tap]:
+    class ArgParser(Tap):
+        # Overwriting configure would force a user to remember to call super().configure if they want to overwrite it
+        # Instead, overwrite _configure
+        def _configure(self):
+            for arg_data in args_data:
+                variable = arg_data.name
+                self._annotations[variable] = str if arg_data.annotation is Any else arg_data.annotation
+                self.class_variables[variable] = {"comment": arg_data.description or ""}
+                if arg_data.is_required:
+                    kwargs = {}
+                else:
+                    kwargs = dict(required=False, default=arg_data.default)
+                self.add_argument(f"--{variable}", **kwargs)
+
+            super()._configure()
+
+    return ArgParser
+
+
+def convert_to_tap_class(class_or_function: _ClassOrFunction, **func_kwargs) -> Type[Tap]:
```

Comment:
nit: since this might be used in something like—

```python
class CustomTap(convert_to_tap_class(MyDataModel)):
    pass
```

—would it be better to name it something shorter like `to_tap_class` or `tap_class`?

Reply:
Thank you for the suggestion! I prefer `to_tap_class`!